test/adapters/IDB_test.js

```diff
@@ -19,6 +20,102 @@ describe("adapter.IDB", () => {
 
     afterEach(() => sandbox.restore());
 
+    describe("#batch", () => {
+      describe("Succesful transaction", () => {
+        function successfulBatch() {
+          return db.batch(batch => {
+            batch.create({id: 1, name: "foo"});
+            batch.create({id: 2, name: "bar"});
+          });
+        }
+
+        it("should resolve with the list of successful operations", () => {
+          return successfulBatch()
+            .should.eventually.have.property("operations").eql([
+              {type: "create", data: {id: 1, name: "foo"}},
+              {type: "create", data: {id: 2, name: "bar"}},
+            ]);
+        });
+
+        it("should resolve with the list of errors", () => {
+          return successfulBatch()
+            .should.eventually.have.property("errors").eql([]);
+        });
+
+        it("should batch create records", () => {
+          return successfulBatch()
+            .then(_ => db.list())
+            .should.become([
+              {id: 1, name: "foo"},
+              {id: 2, name: "bar"},
+            ]);
+        });
+
+        it("should perform different crud operations in order", () => {
+          return db.create({id: 1, name: "foo"})
+            .then(_ => db.batch(batch => {
+              batch.delete(1);
+              batch.create({id: 2, name: "bar"});
+              batch.update({id: 2, name: "baz"});
+            }))
+            .then(_ => db.list())
+            .should.become([{id: 2, name: "baz"}]);
+        });
+      });
+
+      describe("Failing transaction", () => {
+        it("should expose failing operation errors", () => {
+          return db.batch(batch => {
+            batch.create({id: 1, name: "foo"});
+            batch.create({id: 2, name: "bar"});
+            batch.create(1);
+            batch.create(2);
+          })
+            .then(res => {
+              expect(res.errors).to.have.length.of(2);
+              expect(res.errors[0].error.name).eql("DataError");
```

Comment:
We should really document a good list of different errors that may appear in `SyncResults.errors`, and what they mean.